package/cpp/api/JsiSkImageFilterFactory.h

```diff
@@ -94,6 +94,26 @@ class JsiSkImageFilterFactory : public JsiSkHostObject {
                      getContext(), SkImageFilters::Compose(outer, inner)));
   }
 
+
+  JSI_HOST_FUNCTION(MakeBlend) {
+    auto mode = static_cast<SkBlendMode>(arguments[0].asNumber());
+    sk_sp<SkImageFilter> background = JsiSkImageFilter::fromValue(runtime, arguments[1]);
+    sk_sp<SkImageFilter> foreground = nullptr;
+
+    if(count > 2 && !arguments[2].isNull()) {
+      foreground = JsiSkImageFilter::fromValue(runtime, arguments[2]);
+    }
+
+    SkImageFilters::CropRect cropRect = {};
+    if (count > 3 && !arguments[3].isUndefined()) {
```

Comment:
In the test for arguments[2] we test for `isNull` and here we're testing `isUndefined` - is it done "on purpose"? Just curious if this is a bug or intentional :)

Reply:
yes this is done on purpose.
In the canvas kit the null argument is mandatory (it doesn't need to be) but it's actually quite clean when composing filters via reduce for instance (you cannot shoot yourself with an undefined).